### PR TITLE
fix: fix failing docker CI due to permission issues on PRs

### LINKED ISSUE #93: docker-build fails for DockerHub and Scaleway on PRs from forks
## Context

celestia-app just upgraded to v0.3.1 of this repo. 

## Problem

CI fails for docker-build to DockerHub and Scaleway with:

```
Error: Username and password required
```

Link: https://github.com/celestiaorg/celestia-app/actions/runs/7789514729/job/21241591600?pr=3071#step:3:11

## Proposal

Skip this task if the workflow is expected to fail. I think it's expected to fail for PRs created from forks cc: @MSevey

### PR DESCRIPTION
**Pull request by MSevey**
## Overview

Closes #93 

This change skips running the dockerhub and scaleway runs when it is a PR. 
We use org level secrets for logging into dockerhub and scaleway, and these secretes are only accessible by native branches, or users with write access. 
This means that on PRs from forks or non team members, the runs would fail. 
These still pass on main on merge, because at that point, the permission issue is no longer present. 

Additional this change removes the appended hash to the action run, which causes issues with linking the action to the branch protection rules. 

Verified and tested: https://github.com/celestiaorg/.github/actions/runs/7790602743

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by rootulp** [comment]
[nit] I think the PR title may have meant to use the conventional commit prefix `fix:` instead of `bug:`. Ref: https://www.conventionalcommits.org/en/v1.0.0/

File: .github/workflows/dockerfile_workflow_test.yaml [hunk 1]
```diff
@@ -1,6 +1,16 @@
+#
+# This workflow is used to test the `reusable_dockerfile_pipeline` action used
+# to build and push the Docker image to the container registries.
+#
+# The reason this workflow targets the develop branch is so that we can test the
+# action in the PR. If we targeted main, we would need to merge changes into
+# before being able to test them.
```
Comment (rootulp) [inline .github/workflows/dockerfile_workflow_test.yaml:7]: merge change into what?

File: .github/workflows/reusable_dockerfile_pipeline.yml [hunk 2]
```diff
@@ -181,35 +181,48 @@ jobs:
...
             user-secret: SCALEWAY_USERNAME
             token-secret: SCW_SECRET_KEY
             registry-url: rg.fr-par.scw.cloud
             registry-owner: celestiaorg
+            run-on-pr: "false"
       fail-fast: false
     steps:
+      - name: Check run conditions
+        id: run_check
+        # We only want to run when the registry is able to run on pr or if it is a merge event
```
Comment (rootulp) [inline .github/workflows/reusable_dockerfile_pipeline.yml:209]:
```suggestion
        # We only want to run when the registry is able to run on PR or if it is a merge event
```

**Review by rootulp — DISMISSED** [review]
Untested but roughly LGTM

File: .github/workflows/dockerfile_workflow_test.yaml (hunk 1 above)
Comment (MSevey) [reply .github/workflows/dockerfile_workflow_test.yaml:7]: into main

File: .github/workflows/dockerfile_workflow_test.yaml [hunk 3]
```diff
@@ -1,6 +1,16 @@
+#
+# This workflow is used to test the `reusable_dockerfile_pipeline` action used
+# to build and push the Docker image to the container registries.
+#
+# The reason this workflow targets the develop branch is so that we can test the
+# action in the PR. If we targeted main, we would need to merge changes into
```
Comment (MSevey) [inline .github/workflows/dockerfile_workflow_test.yaml:6]:
```suggestion
# action in the PR. If we targeted main, we would need to merge changes in
```

**Review by Bidon15 — DISMISSED** [review]

File: .github/workflows/dockerfile_workflow_test.yaml (hunk 1 above)
Comment (rootulp) [reply .github/workflows/dockerfile_workflow_test.yaml]: TBH I think the grammar here is still incorrect. Proposal to clarify in the comment:

```
If we targeted main, we would need to merge changes into main before being able to test them.
```

File: .github/workflows/dockerfile_workflow_test.yaml [hunk 4]
```diff
@@ -1,6 +1,16 @@
+#
+# This workflow is used to test the `reusable_dockerfile_pipeline` action used
+# to build and push the Docker image to the container registries.
+#
+# The reason this workflow targets the develop branch is so that we can test the
+# action in the PR. If we targeted main, we would need to merge changes in
```
Comment (MSevey) [inline .github/workflows/dockerfile_workflow_test.yaml:6]:
```suggestion
# action in the PR. If we targeted main, we would need to merge changes into main
```

**Review by Bidon15 — APPROVED** [review]

**Review by sysrex — APPROVED** [review]
lgtm

**Pull request merged**